pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py

```diff
@@ -0,0 +1,255 @@
+    """
+
+    def test_all(self):
+        """Publish the rpm rsync distributor before the yum distributor."""
+        # Create a user. Save its private key where Pulp can find it.
+        ssh_user, priv_key = self.make_user(self.cfg)
+        ssh_identity_file = self.write_private_key(self.cfg, priv_key)
+
+        # Create a repository with an importer and pair of distributors, and
+        # sync it.
+        api_client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        body['distributors'].append({
+            'distributor_id': utils.uuid4(),
+            'distributor_type_id': 'rpm_rsync_distributor',
+            'distributor_config': {
+                'predistributor_id': body['distributors'][0]['distributor_id'],
+                'remote': {
+                    'host': urlparse(self.cfg.base_url).netloc,
+                    'root': '/home/' + ssh_user,
+                    'ssh_identity_file': ssh_identity_file,
+                    'ssh_user': ssh_user,
+                }
+            }
+        })
+        repo_href = api_client.post(REPOSITORY_PATH, body)['_href']
+        self.resources.add(repo_href)
+
+        # Publish with the rpm rsync distributor
+        with self.assertRaises(exceptions.TaskReportError):
+            api_client.post(urljoin(repo_href, 'actions/publish/'), {
+                'id': body['distributors'][1]['distributor_id']
+            })
+
+        # Verify that the rsync distributor hasn't placed files
+        sudo = '' if utils.is_root(self.cfg) else 'sudo '
+        cmd = (sudo + 'ls -1 /home/{}'.format(ssh_user)).split()
+        dirs = set(cli.Client(self.cfg).run(cmd).stdout.strip().split('\n'))
+        self.assertNotIn('content', dirs)
+
+
+class PublishTestCase(_MakeUserMixin, utils.BaseAPITestCase):
+    """Publish a repository with the rsync distributor.
+
+    Do the following:
+
+    1. Create a repository with a yum distributor and rsync distributor. Add
+       content units to the repository.
+    2. Publish with the yum distributor.
+    3. Publish with the rpm rsync distributor. Verify that the correct files
+       are in the target directory.
+    """
+
+    def test_all(self):
+        """Publish a repository several times with the rsync distributor."""
+        # Create a user. Save its private key where Pulp can find it.
+        ssh_user, priv_key = self.make_user(self.cfg)
+        ssh_identity_file = self.write_private_key(self.cfg, priv_key)
+
+        # Create a repository with an importer and pair of distributors, and
+        # sync it.
+        api_client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        body['distributors'].append({
```

Comment:
This same log was used in the previous test case, we may consider creating a helper function if this is required for any other test case.